addons/hr/models/hr_department.py

```diff
@@ -106,7 +101,12 @@ def write(self, vals):
             manager_id = vals.get("manager_id")
             # set the employees's parent to the new manager
             self._update_employee_manager(manager_id)
-        return super().write(vals)
+
+        try:
```

Comment:
A UserError could be raised for much more different reason than this parent/child mismatch. 

Revert and keep original behavior. If the generic error message from parent_store is a problem, this could be addressed in another task.